src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -75,15 +75,25 @@ class TosService (val directoryDao: DirectoryDAO, val registrationDao: Registrat
     *         IO[Some(false)] if ToS is enabled and the user hasn't accepted
     *         IO[None] if ToS is disabled
     */
-  def getTosStatus(user: WorkbenchSubject, samRequestContext: SamRequestContext): IO[Option[Boolean]] = {
+  def getTosStatus(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Option[Boolean]] = {
     if (tosConfig.enabled) {
-      getTosGroupName(samRequestContext).flatMap {
-        case Some(groupName) => directoryDao.isGroupMember(groupName, user, samRequestContext).map(Option(_))
-        case None => IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Terms of Service group ${getGroupNameString()} not found.")))
-      }
+      directoryDao.isGroupMember(getGroupName(), userId, samRequestContext).map(Option(_))
     } else IO.none
   }
 
+  /**
+    * If grace period enabled, don't check ToS, return true
+    * If ToS disabled, return true
+    * Otherwise return true if user has accepted ToS
+    */
+  def isTermsOfServiceStatusAcceptable(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Boolean] = {
+    if (tosConfig.isGracePeriodEnabled) {
```

Comment:
What is the grace period, and when would this be true?

Reply:
It is a config setting. I think it means that users asked to accept the new ToS but Terra is not yet enforcing it.